File: 01_basics/01_04_reverse_numbers/reverse.py
def reverseNumber(x: int):
    sign = 1 if x > 0 else -1
    reverse = sign * int(str(abs(x))[::-1])
    return reverse if -2**31 < reverse < 2**31 else 0

number = 1234
print(f"Reverse of {number} is {reverseNumber(number)}")

File: 01_basics/01_04_reverse_numbers/reverse.cpp
#include <iostream>
#include <limits>

int reverseNumber(int x) {
    int sign = (x > 0) ? 1 : -1;
    long long reverse = sign * stoll(std::to_string(std::abs(x)));
    return (reverse < std::numeric_limits<int>::min() || reverse > std::numeric_limits<int>::max()) ? 0 : reverse;
}

int main() {
    int number = 1234;
    std::cout << "Reverse of " << number << " is " << reverseNumber(number) << std::endl;
    return 0;
}
